zap/src/parser/reports.rs

```diff
@@ -227,6 +235,15 @@ impl<'src> Report<'src> {
 			Self::AnalyzeInvalidOptionalType { span, .. } => {
 				vec![Label::primary((), span.clone()).with_message("must be removed")]
 			}
+
+			Self::AnalyzeUnboundedRecursiveType {
+				decl_span, use_span, ..
+			} => {
+				vec![
+					Label::primary((), decl_span.clone()).with_message("declared here"),
+					Label::secondary((), use_span.clone()).with_message("used recursively here"),
```

Comment:
Where it is used recursively should be the primary issue
```suggestion
					Label::secondary((), decl_span.clone()).with_message("declared here"),
					Label::primary((), use_span.clone()).with_message("used recursively here"),
```